plugin/discovery/find.go

```diff
@@ -0,0 +1,167 @@
+package discovery
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const machineName = runtime.GOOS + "-" + runtime.GOARCH
+
+// FindPlugins looks in the given directories for files whose filenames
+// suggest that they are plugins of the given kind (e.g. "provider") and
+// returns a PluginMetaSet representing the discovered potential-plugins.
+//
+// Currently this supports two different naming schemes. The current
+// standard naming scheme is a subdirectory called $GOOS-$GOARCH containing
+// files named terraform-$KIND-$NAME-V$VERSION. The legacy naming scheme is
+// files directly in the given directory whose names are like
+// terraform-$KIND-$NAME.
```

Comment:
Should we also support `$GOOS-GOARCH/terraform-$KIND-$NAME` for non-versioned plugins that are at least vendored for the correct os/arch? Or are plugins from here on out _always_ going to have a version?